File: Logic_Building_Problems/3_sum_of_naturals.py
def getSumWithLoop(number):
    sum = 0
    for i in range(number+1):
        sum += i
    return sum

def getSumWithFormula(number):
    """
        This method is more efficient because it doesn't have any loop.
    """
    return (number * (number + 1)) // 2

if __name__ == '__main__':
    input_number = int(input())
    # print(getSumWithLoop(number=input_number))
    print(getSumWithFormula(number=input_number))

File: Logic_Building_Problems/3_sum_of_naturals.cpp
#include <iostream>
using namespace std;

int getSumUsingLoop(int number) {
    int sum=0;
    for (int i=1; i<=number; i++) {
        sum += i;
    }
    return sum;
}


int getSumUsingFormula(int number) {
    return (number * (number + 1)) / 2;
}


int main() {
    int input_number = 0;
    cin >> input_number;

    // cout << getSumUsingLoop(input_number) << endl;
    cout << getSumUsingFormula(input_number) << endl;
}